### ISSUE TITLE
Dependency conflict with Python 3.12

### ISSUE THREAD

**Issue by simonepignotti** [Bubobubobubobubo/sardine#251]
Hi, thanks for developing and maintaining this great framework!

It looks like it's currently impossible to install on Pyhton 3.12 due to the `LinkPython-extern` dependency, I would suggest temporarily adding Python<3.12 to the pypi specs.

```
 pip install sardine-system
Collecting sardine-system
  Downloading sardine_system-0.4.0-py3-none-any.whl.metadata (5.5 kB)
Collecting ziffers>=0.0.1 (from sardine-system)
  Downloading ziffers-0.0.1-py3-none-any.whl (41 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 41.3/41.3 kB 1.3 MB/s eta 0:00:00
Collecting parsimonious>=0.10.0 (from sardine-system)
  Downloading parsimonious-0.10.0-py3-none-any.whl (48 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 48.4/48.4 kB 3.9 MB/s eta 0:00:00
Collecting Click>=8.1.3 (from sardine-system)
  Downloading click-8.1.7-py3-none-any.whl.metadata (3.0 kB)
INFO: pip is looking at multiple versions of sardine-system to determine which version is compatible with other requirements. This could take a while.
Collecting sardine-system
  Downloading sardine_system-0.2.1-py3-none-any.whl (100 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100.3/100.3 kB 7.8 MB/s eta 0:00:00
Collecting appdirs>=1.4 (from sardine-system)
  Downloading appdirs-1.4.4-py2.py3-none-any.whl (9.6 kB)
Collecting easing-functions>=1.0.4 (from sardine-system)
  Downloading easing_functions-1.0.4-py3-none-any.whl (15 kB)
Collecting exceptiongroup>=1.0.4 (from sardine-system)
  Downloading exceptiongroup-1.2.0-py3-none-any.whl.metadata (6.6 kB)
Collecting inquirerpy>=0.3.4 (from sardine-system)
  Downloading InquirerPy-0.3.4-py3-none-any.whl (67 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 67.7/67.7 kB 6.2 MB/s eta 0:00:00
Collecting lark>=1.1 (from sardine-system)
  Downloading lark-1.1.8-py3-none-any.whl.metadata (1.9 kB)
ERROR: Cannot install sardine-system==0.2.1 and sardine-system==0.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    sardine-system 0.4.0 depends on LinkPython-extern>=1.0.2
    sardine-system 0.2.1 depends on linkpython-extern>=1.0.2

To fix this you could try to:
1. loosen the range of package versions you've specified
2. remove package versions to allow pip attempt to solve the dependency conflict

ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/topics/dependency-resolution/#dealing-with-dependency-conflicts
```

**Comment by simonepignotti** [comment]
Alternatively, you could bump LinkPython-extern to [1.0.4a1](https://github.com/thegamecracks/link-python?tab=readme-ov-file#newer-python-support):
>v1.0.4a1 adds support for Python 3.12 after updating pybind11 to 2.11.1. Pre-built wheels are available on PyPI. This version must be installed using pip install LinkPython-extern==1.0.4a1.

**Comment by Bubobubobubobubo** [comment]
Thanks for bringing up the issue! I am still using Python 3.11. I'll update as soon as possible.

**Comment by Bubobubobubobubo** [comment]
@thegamecracks just released the `1.0.4`. I bumped the `pyproject.toml`. It should be fixed now!

**Comment by simonepignotti** [comment]
Great, thanks a lot for the quick fix!
